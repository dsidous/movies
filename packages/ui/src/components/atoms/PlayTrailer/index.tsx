import React, { useState } from 'react';
import ModalVideo from 'react-modal-video';

import { Button } from '@material-ui/core';
import Movie from '@material-ui/icons/Movie';

import useStyles from './styles';
import 'react-modal-video/css/modal-video.min.css';

interface Props {
  video: {
    key: string;
  }
}

const PlayTrailer: React.FC<Props> = ({ video: { key } }) => {
  const classes = useStyles();
  const [isOpen, setIsopen] = useState(false);

  const openModal = () => {
    setIsopen(true);
  };

  return (
    <div className={classes.root}>
      <ModalVideo
        channel="youtube"
        isOpen={isOpen}
        videoId={key}
        onClose={() => setIsopen(false)}
      />
      <Button
        className={classes.playButton}
        onClick={openModal}
        size="small"
        startIcon={<Movie />}
        variant="contained"
      >
        Watch Trailer
      </Button>
    </div>
  );
};

export default PlayTrailer;
