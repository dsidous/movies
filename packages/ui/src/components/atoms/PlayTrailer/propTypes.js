import { object } from 'prop-types';

export const propTypes = {
  video: object,
};

export const defaultProps = {};
