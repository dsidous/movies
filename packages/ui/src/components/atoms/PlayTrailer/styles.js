import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  {
    root: {
      gridArea: 'mtrailer',
      alignSelf: 'flex-end',
      marginRight: 10,
    },

    playButton: {
      background: '#000',
      border: 'none',
      color: '#bbb',
      cursor: 'pointer',
      display: 'inline-block',
      fontSize: 12,
      margin: '20px 0 0',
      padding: '5px 10px',

      '&:hover': {
        color: '#999',
      },

      '& span': {
        margin: '0 5px 0 0',
      },
    },
  },
  { name: 'trailer' },
);
