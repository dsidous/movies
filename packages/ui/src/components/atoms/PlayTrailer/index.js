import React, { useState } from 'react';
import ModalVideo from 'react-modal-video';
import { propTypes } from './propTypes';

import 'react-modal-video/css/modal-video.min.css';

const PlayTrailer = ({ video: { key } }) => {
  const [isOpen, setIsopen] = useState(false);

  const openModal = () => {
    setIsopen(true);
  };

  return (
    <div className="play-trailer-wrapper">
      <ModalVideo
        channel="youtube"
        isOpen={isOpen}
        videoId={key}
        onClose={() => setIsopen(false)}
      />
      <button type="button" className="play-trailer" onClick={openModal}>
        <span>
          <i className="fa fa-angle-right" />
          {' Watch Trailer'}
        </span>
      </button>
    </div>
  );
};

PlayTrailer.propTypes = propTypes;

export default PlayTrailer;
