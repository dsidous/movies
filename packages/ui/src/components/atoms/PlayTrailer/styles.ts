import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      gridArea: 'mtrailer',
      alignSelf: 'flex-end',
      marginRight: theme.spacing(1),
    },

    playButton: {
      background: '#000',
      color: '#bbb',

      '&:hover': {
        background: '#000',
        color: '#fff',
      },
    },
  }),
  { name: 'trailer' },
);
