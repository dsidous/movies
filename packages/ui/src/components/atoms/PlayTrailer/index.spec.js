import React from 'react';
import { shallow } from 'enzyme';
import ModalVideo from 'react-modal-video';

import PlayTrailer from '.';

const mockProps = {
  video: {
    key: 'somekey',
  },
};

describe('Atoms/PlayTrailer', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<PlayTrailer {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should open onClick', () => {
    const wrapper = shallow(<PlayTrailer {...mockProps} />);
    const modal = wrapper.find(ModalVideo);

    wrapper.find('.play-trailer').simulate('click');
    expect(wrapper.find(ModalVideo).props().isOpen).toBe(true);

    modal.simulate('close');
    expect(wrapper.find(ModalVideo).props().isOpen).toBe(false);
  });
});
