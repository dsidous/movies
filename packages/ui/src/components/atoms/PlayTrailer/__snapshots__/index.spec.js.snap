// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/PlayTrailer should render as expected 1`] = `ShallowWrapper {}`;
